main.go

```diff
@@ -70,6 +76,40 @@ func main() {
 	}
 }
 
+//abHandler ... percentA is the percentage of request that handler 'a' is used
+func abHandler(a, b http.Handler, percentA int) http.Handler {
+	if percentA < 0 || percentA > 100 {
+		panic("Percent 'a' but be between 0 and 100")
+	}
+	rand.Seed(time.Now().UnixNano())
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// Detect cookie
+		cookie, _ := req.Cookie("homepage-version")
+
+		if cookie == nil {
+			var cookieValue string
+			if rand.Intn(100) < percentA {
+				cookieValue = "A"
+			} else {
+				cookieValue = "B"
+			}
+
+			expiration := time.Now().Add(365 * 24 * time.Hour)
+			cookie = &http.Cookie{Name: "homepage-version", Value: cookieValue, Expires: expiration}
+			http.SetCookie(w, cookie)
+		}
+
+		// Use cookie value to direct to a or b handler
+		if cookie.Value == "A" {
+			a.ServeHTTP(w, req)
```

Comment:
Not sure about the implementation details of this middleware, so this comment might be void, but should we return after calling `ServeHTTP`?